File: Ruby/kyu-8/_043__calculate-bmi/solution.rb
def bmi(weight, height)
  conditions = [
    ->(bmi) { bmi <= 18.5 ? 'Underweight' : nil },
    ->(bmi) { bmi <= 25 ? 'Normal' : nil },
    ->(bmi) { bmi <= 30 ? 'Overweight' : nil },
    ->(bmi) { 'Obese' },
  ]
  
  bmi_value = weight / (height ** 2)
  
  conditions.find { |condition| condition[bmi_value] }[bmi_value]
end

File: Python/kyu-8/_043__calculate-bmi/solution.py
def bmi(weight, height):
    bmi = weight / height ** 2
    res = 'Obese'
    if bmi <= 30:
        res = 'Overweight'
        if bmi <= 25:
            res = 'Normal'
            if bmi <= 18.5:
                res = 'Underweight'
    return res


print(bmi(50, 1.8))